cmd/kubelet/app/server.go

```diff
@@ -285,6 +281,72 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 	return cmd
 }
 
+// loadKubeletConfiguration serializes an YAML in the internal KubeletConfiguration
+func loadKubeletConfiguration(configFile string, args []string) (*kubeletconfiginternal.KubeletConfiguration, error) {
+	kubeletConfig, err := loadConfigFile(configFile)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	// We must enforce flag precedence by re-parsing the command line into the new object.
+	// This is necessary to preserve backwards-compatibility across binary upgrades.
+	// See issue #56171 for more details.
+	if err := kubeletConfigFlagPrecedence(kubeletConfig, args); err != nil {
+		return nil, err
+	}
+
+	return kubeletConfig, nil
+}
+
+// loadKubeletInstanceConfiguration serializes the instance configuration in the internal KubeletInstanceConfiguration
+func loadKubeletInstanceConfiguration(
+	kubeletConfig *kubeletconfiginternal.KubeletConfiguration,
+	instanceConfigFile string,
+) (*kubeletconfiginternal.KubeletInstanceConfiguration, error) {
+	kubeletInstanceConfig, err := options.NewKubeletInstanceConfiguration(kubeletConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeletInstanceConfig, err = loadInstanceConfigFile(instanceConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	return kubeletInstanceConfig, err
+}
+
+// mergeKubeletConfiguration merges specific fields from KubeletInstanceConfiguration to KubeletConfiguration
+func mergeKubeletConfiguration(kc *kubeletconfiginternal.KubeletConfiguration, kic *kubeletconfiginternal.KubeletInstanceConfiguration) *kubeletconfiginternal.KubeletConfiguration {
+	if kc.RegisterNode != kic.RegisterNode {
+		kc.RegisterNode = kic.RegisterNode
+	}
+	if kc.HostnameOverride != kic.HostnameOverride {
+		kc.HostnameOverride = kic.HostnameOverride
+	}
+	if kc.NodeIP != kic.NodeIP {
```

Comment:
The downside of the approach is to have two places that will need overwrite of the internal values: before (for default) and after the unmarshal config YAML.